constraints/src/main/java/gov/nasa/jpl/aerie/constraints/time/IntervalAlgebra.java

```diff
@@ -1,71 +1,98 @@
 package gov.nasa.jpl.aerie.constraints.time;
 
-/**
- * An algebra for interval-like operations on some underlying type {@code I}.
- *
- * @param <Alg> A type witnessing the uniqueness of any given implementation.
- *     Another implementation should not be able to use this witness,
- *     and any implementations with the same witness must be equivalent.
- *     This allows containers that sort based on this algebra to constrain operations with other such collections,
- *     e.g. ensuring that `addAll()` is only allowed (or is differently optimized) when the same algebra is used.
- *
- *     If the implementation is unique for a type, and is private, the implementing class may witness itself.
- *     If the implementation is public, use a `private enum Witness {}` nested into the implementing class.
- *     If the implementation may vary in finitely many ways, use as many witnesses as there are ways.
- *     If the implementation may vary more broadly, may God have mercy on your soul --
- *     either use an existential, or use something less fine-grained and be very very careful.
- * @param <I> The type on which this algebra supports interval operations.
- */
-public interface IntervalAlgebra<Alg, I> {
-  boolean isEmpty(I x);
-
-  I unify(I x, I y);
-  I intersect(I x, I y);
-
-  I lowerBoundsOf(I x);
-  I upperBoundsOf(I x);
-
-
-  default boolean overlaps(I x, I y) {
+import gov.nasa.jpl.aerie.merlin.protocol.types.Duration;
+
+import static gov.nasa.jpl.aerie.constraints.time.Interval.Inclusivity.Exclusive;
+import static gov.nasa.jpl.aerie.constraints.time.Interval.Inclusivity.Inclusive;
+
+public class IntervalAlgebra {
+
+  private Interval horizon;
+
+  public IntervalAlgebra() {
+    horizon = Interval.between(Duration.MIN_VALUE, Duration.MAX_VALUE);
+  }
+
+  public IntervalAlgebra(Interval horizon) {
+    this.horizon = horizon;
+  }
+
+  public final boolean isEmpty(Interval x) {
+    return x.isEmpty();
+  }
+
+  public final Interval unify(final Interval x, final Interval y) {
+    return Interval.unify(x, y);
+  }
+
+  public final Interval intersect(final Interval x, final Interval y) {
+    return Interval.intersect(x, y);
+  }
+
+  public final Interval lowerBoundsOf(final Interval x) {
+    if (x.isEmpty()) return Interval.FOREVER;
+    return Interval.between(
+        horizon.start,
+        Inclusive, x.start,
+        x.startInclusivity.opposite()
+    );
+  }
+
+  public final Interval upperBoundsOf(final Interval x) {
+    if (x.isEmpty()) return Interval.FOREVER;
+    return Interval.between(
+        x.end,
+        x.endInclusivity.opposite(), horizon.end,
+        Inclusive
+    );
+  }
+
+  public final Interval bottom() {
+    return Interval.between(horizon.start, Exclusive, horizon.start, Exclusive);
```

Comment:
🔶 On first read, I thought that the choice of `horizon.start` here was arbitrary (it seemed to me that `bottom` could be any degenerate interval).

However, `setInternal` makes assumptions about the value of `bottom()`. Can we make those assumptions explicit in an `INVARIANT` comment here?

Reply:
Due to discussions with Jonathan, I am removing `IntervalAlgebra.horizon` entirely. Truncating Windows to the plan bounds is not the Windows responsibility; we will have a `select` function available so that any Windows consumers can do that themselves if they want.

Reply:
removed